lua/entities/gmod_wire_dhdd.lua

```diff
@@ -14,11 +14,33 @@ function ENT:Initialize()
 	self.Outputs = WireLib.CreateOutputs( self, { "Memory [ARRAY]", "Size" } )
 	self.Inputs = WireLib.CreateInputs( self, { "Data [ARRAY]", "Clear", "AllowWrite" } )
 
-	self.Memory = {}
+	self:SetMemory()
 	self.ROM = false
 	self.AllowWrite = true
+end
+
+function ENT:SetMemory(Memory)
```

Comment:
Does the proxy actually need to be recreated each time you want to change `self.Memory`? I don't see why it would, as all the accesses to it are via `self.Memory` (which is a field lookup) rather than the local `Memory`.